src/Microsoft.DotNet.ProjectModel/Files/IncludeEntry.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace Microsoft.DotNet.ProjectModel.Files
+{
+    public class IncludeEntry : IEquatable<IncludeEntry>
+    {
+        public string TargetPath { get; }
+
+        public string SourcePath { get; }
+
+        public bool IsCustomTarget { get; set; }
+
+        public IncludeEntry(string target, string source)
+        {
+            TargetPath = target;
+            SourcePath = source;
+        }
+
+        public override bool Equals(object obj)
+        {
+            return Equals((IncludeEntry)obj);
+        }
+
+        public override int GetHashCode()
+        {
+            return base.GetHashCode();
```

Comment:
You have to over ride `GetHashCode` if `Equals` is overridden. To my knowledge the hash code for `IncludeEntry` is not critical since it is not used as key in dictionary or as an element in HashSet. This should be fine.